src/document.rs

```diff
@@ -54,46 +54,44 @@ impl Document {
     ///
     /// This function will panic if the path contains a non UTF-8 character
     #[must_use]
-    pub fn swap_filename(filename: &str) -> String {
+    pub fn swap_filename(filename: &str) -> PathBuf {
```

Comment:
Could we make it so that `filename` here is a `PathBuf` as well? This would remove all instances of `filename.as_str().unwrap()`, and `swap_filename` should be easier to read and write if it only deals with `Path` s.

Reply:
Sure 👍🏾